test/Microsoft.Extensions.DataIngestion.Tests/Chunkers/SectionChunkerTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.DataIngestion.Chunkers;
+using Microsoft.ML.Tokenizers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Extensions.DataIngestion.Tests.Chunkers
+{
+    public class SectionChunkerTests : DocumentChunkerTests
+    {
+        protected override IDocumentChunker CreateDocumentChunker()
+        {
+            return new SectionChunker();
+        }
+
+        private IDocumentChunker CreateSizeLimitedChunker()
+        {
+            var tokenizer = TiktokenTokenizer.CreateForModel("gpt-4o");
+            return new SectionChunker(512, tokenizer);
+        }
+
+        [Fact]
+        public async Task OneSection()
+        {
+            Document doc = new Document("doc");
+            doc.Sections.Add(new DocumentSection
+            {
+                Elements =
+                {
+                    new DocumentParagraph("This is a paragraph."),
+                    new DocumentParagraph("This is another paragraph.")
+                }
+            });
+            IDocumentChunker chunker = CreateDocumentChunker();
+            List<DocumentChunk> chunks = await chunker.ProcessAsync(doc);
+            Assert.Single(chunks);
+            string expectedResult = "This is a paragraph.\nThis is another paragraph.\n";
```

Comment:
IIRC `string.Join` is not going to append the element at the end

```suggestion
            string expectedResult = "This is a paragraph.\nThis is another paragraph.";
```